src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,155 @@
+
+public class RegistrationServiceTest {
+    private static final int MIN_AGE = 18;
+    private RegistrationServiceImpl registrationService;
+
+    @BeforeEach
+    void setup() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void register_validUser_successfulRegistration() {
+        User user = new User();
+        user.setLogin("username");
+        user.setPassword("password123");
+        user.setAge(20);
+
+        User registeredUser = registrationService.register(user);
+        assertNotNull(registeredUser);
+        assertEquals(user.getLogin(), registeredUser.getLogin());
+    }
+
+    @Test
+    public void register_existingUser_throwException() {
+        User user1 = new User();
+        user1.setLogin("username");
+        user1.setPassword("password123");
+        user1.setAge(20);
+
+        User user2 = new User();
+        user2.setLogin("username");
+        user2.setPassword("anotherpassword");
+        user2.setAge(25);
+
+        Exception exception = assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+            registrationService.register(user2);
+        });
+
+        String expectedMessage = "User with this login already exists";
+        String actualMessage = exception.getMessage();
+
+        assertTrue(actualMessage.contains(expectedMessage));
+    }
+
+    @Test
+    public void register_shortLogin_throwException() {
+        User user = new User();
+        user.setLogin("short");
+        user.setPassword("password123");
+        user.setAge(20);
+
+        Exception exception = assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user));
+
+        String expectedMessage = "Login and password must be at least 6 characters long";
+        String actualMessage = exception.getMessage();
+
+        assertTrue(actualMessage.contains(expectedMessage));
+    }
+
+    @Test
+    public void register_shortPassword_throwException() {
+        User user = new User();
+        user.setLogin("username");
+        user.setPassword("short");
+        user.setAge(20);
+
+        Exception exception = assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user));
+
+        String expectedMessage = "Login and password must be at least 6 characters long";
+        String actualMessage = exception.getMessage();
+
+        assertTrue(actualMessage.contains(expectedMessage));
+    }
+
+    @Test
+    public void register_lowAge_throwException() {
+        User user = new User();
+        user.setLogin("username");
+        user.setPassword("password123");
+        user.setAge(17);
+
+        Exception exception = assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user));
+
+        String expectedMessage = "Not valid age: "
+                + user.getAge()
+                + ". Min allowed age is " + MIN_AGE;
```

Comment:
there is no sense to do some concatenation - you should already know which message to expect